vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/implementation/VLogReasoner.java

```diff
@@ -808,4 +813,59 @@ ReasonerState getReasonerState() {
 	void setReasonerState(ReasonerState reasonerState) {
 		this.reasonerState = reasonerState;
 	}
+
+	@Override
+	public Correctness writeInferences(OutputStream stream) throws IOException {
+		QueryResult queryAnswer;

```

Comment:
why declare this variable here and change its value, instead as locally in the try block? It is not necessary and also makes code more unreadable.
Also, variables should be final unless it is absolutely necessary to change their values. 
Reasoner#answerQuery() declares variable `stringQueryResultIterator ` that is only initialised in a try block, as this variable is later returned. Here,  you don't need the value of `queryAnswer ` outside the try block.